build.gradle

```diff
@@ -24,7 +24,15 @@ dependencies {
     implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.13.3'
 
     testImplementation gradleTestKit()
-    testImplementation 'org.spockframework:spock-core:2.2-groovy-4.0'
+
+    testImplementation(platform('org.apache.groovy:groovy-bom:4.0.4'))
+    testImplementation('org.spockframework:spock-core:2.2-groovy-4.0') {
+        exclude(group: 'org.codehaus.groovy')
```

Comment:
Shouldn't we stick instead to `2.2-groovy-3.0` ? I guess dependabot is not able to see this version as the next one

Reply:
Agreed, I more looking at the implication of overriding Gradle's built-in Groovy version

Reply:
If you are building a plugin, then I'd suggest to stick with Gradle's Groovy version. Otherwise, it doesn't really matter and you can use whatever you want.

Reply:
Absolutely, this is a plugin, this PR won't get merged but I was curious on how to handle it and if there was another approach than using `spock.iKnowWhatImDoing.disableGroovyVersionCheck`

Reply:
I don't quite understand how `spock.iKnowWhatImDoing.disableGroovyVersionCheck` comes into play here? It is only useful when you are want to test Spock with not officially supported Groovy versions, otherwise just use the matching Spock version.